tests/mongo-tools/dump_restore_test.go

```diff
@@ -21,57 +21,100 @@ import (
 	ctx, db := common.Setup(t)
 
-	// TODO restart existing dump from https://github.com/mcampo2/mongodb-sample-databases instead
-	_, err := db.Collection("test").InsertOne(ctx, bson.D{{"foo", "bar"}})
+	dbName := "sample_geospatial"
+	db = db.Client().Database(dbName)
+
+	// cleanup database
+	err := db.Drop(ctx)
 	require.NoError(t, err)
 
-	localRoot := filepath.Join("..", "..", "dumps")
-	containerRoot := "/dumps/"
+	localActualPath := filepath.Join("..", "..", "dumps", "actual")
+	containerActualPath := filepath.Join("/dumps", "actual")
 
-	expectedState := getDatabaseState(t, ctx, db)
+	containerExpectedPath := filepath.Join("/dumps", "expected")
+	localExpectedPath := filepath.Join("..", "..", containerExpectedPath)
+
+	containerSourcePath := filepath.Join("/dumps", "mongodb-sample-databases", "dump")
 
 	// pre-create directories to avoid permission issues
-	err = os.Chmod(localRoot, 0o777)
+	err = os.Chmod(localActualPath, 0o777)
 	require.NoError(t, err)
-	err = os.RemoveAll(filepath.Join(localRoot, db.Name()))
+	err = os.Chmod(localExpectedPath, 0o777)
 	require.NoError(t, err)
-	err = os.Mkdir(filepath.Join(localRoot, db.Name()), 0o777) // 0o777 is typically downgraded to 0o755 by umask
+
+	err = os.RemoveAll(filepath.Join(localActualPath, dbName))
 	require.NoError(t, err)
-	err = os.Chmod(filepath.Join(localRoot, db.Name()), 0o777) // fix after umask
+	err = os.RemoveAll(filepath.Join(localExpectedPath, dbName))
 	require.NoError(t, err)
 
-	// dump a database
-	err = runDockerComposeCommand(
-		"mongodump",
-		"--out", containerRoot,
-		"--db", db.Name(),
-		"--verbose",
-		"mongodb://host.docker.internal:27017/",
-	)
+	err = os.Mkdir(filepath.Join(localActualPath, dbName), 0o777) // 0o777 is typically downgraded to 0o755 by umask
+	require.NoError(t, err)
+	err = os.Mkdir(filepath.Join(localExpectedPath, dbName), 0o777)
+	require.NoError(t, err)
+
+	err = os.Chmod(filepath.Join(localActualPath, dbName), 0o777) // fix after umask
 	require.NoError(t, err)
+	err = os.Chmod(filepath.Join(localExpectedPath, dbName), 0o777)
+	require.NoError(t, err)
+
+	// restore a database from a sample dump
+	mongorestore(t, dbName, containerSourcePath)
+
+	expectedState := getDatabaseState(t, ctx, db)
+
+	// "bootstrap" the expected dump from restored database
+	mongodump(t, dbName, containerExpectedPath)
 
 	// cleanup database
-	ctx, db = common.Setup(t)
+	require.NoError(t, db.Drop(ctx))
+
+	// restore a database from the expected dump
+	mongorestore(t, dbName, containerExpectedPath)
+
+	// dump a database
+	mongodump(t, dbName, containerActualPath)
+
+	// get database state after restore and compare it
+	actualState := getDatabaseState(t, ctx, db)
+	assert.Equal(t, expectedState, actualState)
+
+	// compare dump files. Metadata files are not compared because they
+	// contain different uuid field on every dump
+	compareDirs(t, filepath.Join(localExpectedPath, dbName), filepath.Join(localActualPath, dbName), `\\*.metadata.json`)
```

Comment:
Do want to check that `metadata.json` exist in both localActualPath and localExpectedPath?

Reply:
*.metadata.json is the file containing index definitions and it's not required to perform a dump.
Yes, we could check at least if it exists, because it's actually created on every dump (with a different uuid field), although we don't check how `mongodump` works, but if it works with FerretDB so imo we cand just skip it as we don't even provide the index functionality

Reply:
thanks for the explanation. Yes we ignore index for now. Didn't realise metadata was related to index :see_no_evil: